packages/gestalt/src/Masonry/dynamicHeightsUtils.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * Util functions used to update positions when an item changes the height dynamically
+ */
+import { Cache } from './Cache';
+import { ColumnSpanConfig } from './multiColumnLayout';
+import { Position } from './types';
+
+function shouldExtendArea(area: { top: number; left: number; right: number }, position: Position) {
+  return position.left < area.left || position.left + position.width > area.right;
+}
+
+function isBelowArea(area: { top: number; left: number; right: number }, position: Position) {
+  return (
+    position.left < area.right &&
+    position.left + position.width > area.left &&
+    position.top > area.top
+  );
+}
+
+function recalcHeights<T>({
+  items,
+  changedItem,
+  newHeight,
+  positionStore,
+  measurementStore,
+  getColumnSpanConfig,
+}: {
+  items: ReadonlyArray<T>;
+  changedItem: T;
+  newHeight: number;
+  positionStore: Cache<T, Position>;
+  measurementStore: Cache<T, number>;
+  getColumnSpanConfig?: (item: T) => ColumnSpanConfig;
+}) {
+  const changedItemPosition = positionStore.get(changedItem);
+
+  if (!changedItemPosition || changedItemPosition.height === newHeight) {
+    return;
+  }
+
+  measurementStore.set(changedItem, newHeight);
+
+  const { top, left, width, height } = changedItemPosition;
+  const itemsWithPositions = items.filter((item) => item && positionStore.has(item));
```

Comment:
we could simplify this a bit if we just filter out items with positions below the changed item upfront

`position.top > (changedItemPosition.top + changedItemPosition.height)`

Reply:
I did the change to work only on the items that are after item that changed height. I only filtered by `position.top > changedItemPosition.top` because on the case that the height change is large it can happen that the other logic excludes some small items